portal-api/src/v1/users/sign-in-link.controller.js

```diff
@@ -14,49 +15,104 @@ class SignInLinkController {
     try {
       const { userId, signInToken } = req.params;
 
-      const isValidSignInToken = await this.#signInLinkService.isValidSignInToken({ userId, signInToken });
-
-      if (!isValidSignInToken) {
-        throw new InvalidSignInTokenError(userId);
+      if (req.user._id.toString() !== userId) {
+        throw new InvalidUserIdError(userId);
       }
 
-      await this.#signInLinkService.deleteSignInToken(userId);
+      const signInTokenStatus = await this.#signInLinkService.getSignInTokenStatus({ userId, signInToken });
 
-      const { user, tokenObject } = await this.#signInLinkService.loginUser(userId);
+      switch (signInTokenStatus) {
+        case SIGN_IN_LINK.STATUS.NOT_FOUND: {
+          return res.status(HttpStatusCode.NotFound).json({
+            message: 'Not Found',
+            errors: [
+              {
+                msg: `No matching token for user with id ${req.params.userId}`,
+              },
+            ],
+          });
+        }
+        case SIGN_IN_LINK.STATUS.EXPIRED: {
+          return res.status(HttpStatusCode.Forbidden).json({
+            message: 'Forbidden',
+            errors: [
+              {
+                cause: HTTP_ERROR_CAUSES.TOKEN_EXPIRED,
+                msg: `The provided token is no longer valid for user with id ${req.params.userId}`,
+              },
+            ],
+          });
+        }
+        case SIGN_IN_LINK.STATUS.VALID: {
+          const { user, tokenObject } = await this.#signInLinkService.loginUser(userId);
 
-      return res.status(200).json({
-        success: true,
-        token: tokenObject.token,
-        user: sanitizeUser(user),
-        loginStatus: LOGIN_STATUSES.VALID_2FA,
-        expiresIn: tokenObject.expires,
-      });
+          return res.status(HttpStatusCode.Ok).json({
+            success: true,
+            token: tokenObject.token,
+            user: sanitizeUser(user),
+            loginStatus: LOGIN_STATUSES.VALID_2FA,
+            expiresIn: tokenObject.expires,
+          });
+        }
+
+        default:
+          throw InvalidSignInTokenError(signInToken);
+      }
     } catch (e) {
       console.error(e);
```

Comment:
Can `console.error` please adhere to standard `console.error(''... %s, error)` convention?